python/cudf/cudf/_lib/groupby.pyx

```diff
@@ -62,19 +66,44 @@ _DECIMAL_AGGS = {"COUNT", "SUM", "ARGMIN", "ARGMAX", "MIN", "MAX", "NUNIQUE",
 # workaround for https://github.com/cython/cython/issues/3885
 ctypedef const scalar constscalar
 
+
+cdef _agg_result_from_columns(
+    vector[libcudf_groupby.aggregation_result]& c_result_columns,
+    set column_included,
+    int n_input_columns
+):
+    """Construct the list of result columns from libcudf result. The result
+    contains the same number of lists as the number of input columns. Result
+    for an input column that has no applicable aggregations is an empty list.
+    """
+    cdef int i, j, result_index = 0, n_res_cols = c_result_columns.size()
+    result_columns = []
+    cdef vector[unique_ptr[column]]* c_result
```

Comment:
Regarding the cdefs, it may also help to use [`cdef` blocks](https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html#grouping-multiple-c-declarations). I don't see them often and I suspect not many are aware of them, but I have used them a decent bit before and I think they help readability even if they don't really save any space.

Reply:
I tend to prefer the repeated cdef approach over cdef blocks, because I don’t like adding an indentation level (might imply a different scope or some kind of magic behavior to readers who aren’t familiar). This is a very weak preference so do whatever you like.

Reply:
Opted for cdef blocks, we can come back to revisit cython code formatting as a whole.